### PR TITLE
Some Fixings

### LINKED ISSUE #10939: Tag links broken on other networks (Mastodon, Hubzilla) when posting from API
- [X] I have searched open and closed issues for duplicates

### Bug Description

When post contains hashtags and is made from API, hashtag link is broken on other networks - it contains "https://" prefix twice, e.g. `https://https//friends.deko.cloud/search?tag=test`

### Steps to Reproduce

1. Make a post containing hashtags using Friendica frontend and another using API, e.g. `/usr/bin/curl -u user:pass https://friends.deko.cloud/api/v1/statuses -d source="Friendica" -d status="Post with #test tag."`
2.  Look at the tag link on Mastodon, Hubzilla or Pleroma (presumably any AP network).

**Actual Result:**

Post made from Friendica frontend:
Original - https://friends.deko.cloud/display/d0446be5-4661-79b2-100d-119745727637
Mastodon - https://lor.sh/web/statuses/107175315017641354
Hubzilla - https://hub.deko.cloud/display/b64.aHR0cHM6Ly9mcmllbmRzLmRla28uY2xvdWQvb2JqZWN0cy9kMDQ0NmJlNS00NjYxLTc5YjItMTAwZC0xMTk3NDU3Mjc2Mzc

Post made from Friendica API:
Original - https://friends.deko.cloud/display/d0446be5-1561-79a6-2391-a6b979143900
Mastodon - https://lor.sh/web/statuses/107175126379365976
Hubzilla - https://hub.deko.cloud/display/b64.aHR0cHM6Ly9mcmllbmRzLmRla28uY2xvdWQvb2JqZWN0cy9kMDQ0NmJlNS0xNTYxLTc5YTYtMjM5MS1hNmI5NzkxNDM5MDA

**Expected Result:**

Tag links should be correct.

Friendica Version:
2021.09 stable (git)

### PR DESCRIPTION
**Pull request by nupplaphil**
- Updates messages.po & database.sql (version update for 2021.12-rc)
- Fix BBCode::expandTags()
- Make tests work again

Extracted from #11038 to make it more readable

### CODE REVIEW

**Comment by annando** [comment]
I just realized, that `DI::apiResponse()->exit(...);` doesn't exit anymore.

**Comment by nupplaphil** [comment]
ah yes, thanks!

[Edit] It still exists, but ist redundant with `$this->response->exit(...);` (yet)

File: tests/src/Module/Api/ApiTest.php [hunk 1]
```diff
@@ -51,9 +53,13 @@ protected function setUp(): void
 		parent::setUp(); // TODO: Change the autogenerated stub
 
 		$this->dice = $this->dice
-			->addRule(Authentication::class, ['instanceOf' => AuthenticationDouble::class, 'shared' => true]);
+			->addRule(Authentication::class, ['instanceOf' => AuthenticationDouble::class, 'shared' => true])
+			->addRule(App::class, ['instanceOf' => AppDouble::class, 'shared' => true]);
 		DI::init($this->dice);
 
+		// Manual overwrite for API testrs :-)
```
Comment (MrPetovan) [inline tests/src/Module/Api/ApiTest.php]:
```suggestion
		// Manual override to bypass API authentication
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by annando** [comment]
@nupplaphil it exists, but it doesn't exits.

**Pull request merged**